src/Microsoft.ML.TorchSharp/NasBert/TextClassificationTrainer.cs

```diff
@@ -0,0 +1,857 @@
+            metadata.Add(new SchemaShape.Column(AnnotationUtils.Kinds.KeyValues, SchemaShape.Column.VectorKind.Vector,
+                TextDataViewType.Instance, false));
+
+            outColumns[_options.OutputColumnName] = new SchemaShape.Column(_options.OutputColumnName, SchemaShape.Column.VectorKind.Scalar,
+                    NumberDataViewType.UInt32, true, new SchemaShape(metadata.ToArray()));
+
+            return new SchemaShape(outColumns.Values);
+        }
+
+        private void CheckInputSchema(SchemaShape inputSchema)
+        {
+            // Verify that all required input columns are present, and are of the same type.
+            if (!inputSchema.TryFindColumn(_options.Sentence1ColumnName, out var sentenceCol))
+                throw _host.ExceptSchemaMismatch(nameof(inputSchema), "sentence", _options.Sentence1ColumnName);
+            if (sentenceCol.ItemType != TextDataViewType.Instance)
+                throw _host.ExceptSchemaMismatch(nameof(inputSchema), "sentence", _options.Sentence1ColumnName,
+                    TextDataViewType.Instance.ToString(), sentenceCol.GetTypeString());
+
+            if (!inputSchema.TryFindColumn(_options.LabelColumnName, out var labelCol))
+                throw _host.ExceptSchemaMismatch(nameof(inputSchema), "label", _options.LabelColumnName);
+            if (labelCol.ItemType != NumberDataViewType.UInt32)
+                throw _host.ExceptSchemaMismatch(nameof(inputSchema), "label", _options.LabelColumnName,
+                    NumberDataViewType.UInt32.ToString(), labelCol.GetTypeString());
+
+            if (_options.Sentence2ColumnName != default)
+            {
+                if (!inputSchema.TryFindColumn(_options.Sentence2ColumnName, out var sentenceCol2))
+                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "sentence2", _options.Sentence2ColumnName);
+                if (sentenceCol2.ItemType != TextDataViewType.Instance)
+                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "sentence2", _options.Sentence2ColumnName,
+                        TextDataViewType.Instance.ToString(), sentenceCol2.GetTypeString());
+            }
+        }
+    }
+
+    public sealed class SentenceClassificationTransformer : RowToRowTransformerBase
+    {
+        internal const string LoadName = "NASBERTTrainer";
+        internal const string UserName = "NASBERT Trainer";
+        internal const string ShortName = "NASBERT";
+        internal const string Summary = "NLP with NAS-BERT";
+
+        private readonly Device _device;
+        private readonly SentenceClassificationModel _model;
+        private readonly Vocabulary _vocabulary;
+        private readonly TextClassificationTrainer.Options _options;
+
+        private readonly string _predictedLabelColumnName;
+
+        public readonly SchemaShape.Column SentenceColumn;
+        public readonly SchemaShape.Column SentenceColumn2;
+        public readonly SchemaShape.Column LabelColumn;
```

Comment:
do we usually do these as fields instead of properties?

Reply:
In our `TrainerEstimatorBase` yes so I'll follow that convention here as well.